### DISCUSSION TITLE
Add aggregation (reduction) method for N-steps calculations

### DISCUSSION THREAD

**Discussion by jamartinh** [ymd-h/cpprb#32] [Feature Request]
Hi, would it be possible to add a parameter to the N-step subdict of the ReplayBuffer constructor so that one can specify the kind of aggregator/reductio method of the n-steps returns.

Normally the standard case is accumulate or cumsum, however forinstance for average reinforcement learning would be nice to have not the sum but the mean so one can use N-step returns for average reward RL as well.

I am not being able to compile now cpprb on my machines because my OS is centos 7 and I have compiler obsolescence.

Thanks !

**Comment by ymd-h** [comment]
@jamartinh 
Thank you for your proposal.

Can you give us more detail calculation you want?
Is the reduction applied over future N-steps like ordinary Nstep reward?

For simple average, is it not enough by setting `gamma = 1.0` and dividing rewards by `N` later?

**Comment by jamartinh** [comment]
Hi @ymd-h , do you mean that when creating the N-steps dict for the env_dict I can use for instance:

```python
N=10
n_step_dict = {
    "size": N,
    "gamma": 1.0,
    "rew": "rew",
    "next": "next_obs"}
```

And then, the samples will give be the  sample["rew"]    so I can make just  AVG = sample["rew"]/N ?

It seems that this may work !

**Reply by ymd-h** [reply]
@jamartinh 
Yes, it is what I mean.

If you find any problems, please let us know.

**Comment by jamartinh** [comment]
Thanks!
